StrongInject.Generator/Extensions.cs

```diff
@@ -35,6 +35,22 @@ public static IEnumerable<(T value, int index)> WithIndex<T>(this ImmutableArray
             }
             return value;
         }
+        
+        public static bool TryAdd<TKey, TValue>(this Dictionary<TKey, TValue> dic, TKey key, TValue value, out TValue existingValue)
+        {
+            if (!dic.TryGetValue(key, out existingValue))
+            {
+                dic[key] = value;
+                return true;
+            }
+            return false;
+        }
+        
+        public static void AddOrUpdate<TKey, TValue>(this Dictionary<TKey, TValue> dic, TKey key, TValue value, out TValue existingValue)
```

Comment:
I would have expected a compiler warning to require this to be `out TValue? existingValue`.